packages/shopify-api/lib/types.ts

```diff
@@ -61,3 +61,35 @@ export enum BillingReplacementBehavior {
   ApplyOnNextBillingCycle = 'APPLY_ON_NEXT_BILLING_CYCLE',
   Standard = 'STANDARD',
 }
+
+export enum ValidationType {
+  Flow = 'flow',
+  Webhook = 'webhook',
+}
+
+export const ValidationErrorReason = {
+  MissingBody: 'missing_body',
+  InvalidHmac: 'invalid_hmac',
+  MissingHmac: 'missing_hmac',
+} as const;
```

Comment:
Webhooks and flow currently return slightly different types.

Webhooks returns `missing_headers` and flow returns `missing_hmac`.

With the refactor now webhooks would also return `missing_hmac` if the hmac header specifically is missing, and `missing_headers` if one of the other required headers is missing. (I think this makes sense to pull all of the HMAC specific validations into this new function.)

From my understanding you can't extend enums in js, so that is why I am using a const so I can extend it for webhooks to add `missing_headers`. Let me know if you have another suggestion though!